static/partials/language-specific-deploy/go.md

```diff
@@ -1,137 +1,92 @@
-    ```txt
-    ./
-    src/
-        myapp/
-        foo/
-            module1/
-            module2/
-        module3/
-    ```
-
-    Here you have the modules `myapp`, `foo/module1`, `foo/module2` and `module3`.
-
-2. Create a *clevercloud/go.json* file at the top of your repository:
-
-    ```txt
-    ./
-    clevercloud/
-                go.json
-    src/
-        myapp/
-        ...
-    ```
-
-3. In the go.json file, put the following:
-
-    ```json
-    {
-        "deploy": {
-            "appIsGoPath": true,
-            "main": "myapp"
-        }
-    }
-    ```
-
-    If `appIsGoPath` is present and equals `true`, then we consider that your repo root is the *GOPATH*.
-
-    The `main` field then becomes mandatory and must be the name of the module you want to run.
-
-    E.g. if you want to run `module1`, `main` must be `foo/module1`.
-
-4. (Optional) Add a "execDir" field to the "deploy" object:
-
-    ```json
-    {
-        "deploy": {
-            "appIsGoPath": true,
-            "main": "myapp",
-            "execDir": "src/myapp"
-        }
-    }
-    ```
-
-
-The `execDir` value must be relative to the root of your repo.
-
-In the example above, we will run the application in the src/myapp directory.
+### Mandatory needs
+By default, we consider that your repository contains a single application. Be sure that:
+* It listens to the wild network `0.0.0.0`, not only `localhost` or `127.0.0.1`
+* It listens on port `8080`
+* You follow our build/run instructions
+
+In most cases you won't need to change anything to your application, except host/port and some configuration variables.
+
+### Modern Go project structure
+There are multiple ways to build/run a Go application, and this has evolved over the years. In its modern form a Go project can be a:
+- `Package`: one or more `.go` files you can `build` or `run`. `main` package and `main()` function are the default entry point
+- `Module`: one or more packages you can `install`, defined in a `go.mod` file (`go.sum` for checksums)
+- `Workspace`: one or more modules seamlessly combined, defined in a `go.work` file
+
+A module can be installed locally or from a remote repository if you pass its URL to the `install` command. A `Makefile` is sometimes used to define how a Go project is built, run and/or cleaned. The lightest form of a Go project is a `main.go` file to build. Some years ago, the `src/` folder was often used for source code, but using the `cmd/` folder instead is now a common practice. 
+
+If you want to limit from where a package can be imported it [should placed](https://docs.google.com/document/d/1e8kOo3r51b2BWtTs_1uADIA5djfXhPT36s6eHVRIvaU/edit) in a folder named `ìnternal/`. Access to functions in `.go` files is defined depending [on their name](https://go.dev/tour/basics/3): if it starts with a capital letter it's a public functions, if not it's a private function.
+
+For a complete project, a common files/folders organization can be:
+```
```

Comment:
It might be better to use a file tree here : https://imfing.github.io/hextra/docs/guide/shortcodes/filetree/ 

It's better in terms of UX, interactive and more semantic.

```
{{< filetree/container >}}
  {{< filetree/folder name="application-root/" >}}
    {{< filetree/file name="go.work" >}}
    {{< filetree/file name="Makefile" >}}
    {{< filetree/folder name="cmd/" >}}
      {{< filetree/file name="main.go" >}}
      {{< filetree/file name="other-file.go" >}}
      {{< filetree/file name="other-package.go" >}}
      {{< filetree/file name="…" >}}
    {{< /filetree/folder >}}
        {{< filetree/folder name="module/" >}}
          {{< filetree/file name="go.mod" >}}
          {{< filetree/file name="go.sum" >}}
          {{< filetree/file name="main.go" >}}
          {{< filetree/file name="other-module-file.go" >}}
            {{< filetree/folder name="internal/" >}}
              {{< filetree/folder name="internal-package/" >}}
                {{< filetree/file name="internal-package-file.go" >}}
              {{< /filetree/folder >}}
            {{< /filetree/folder >}}
      {{< /filetree/folder >}}
  {{< /filetree/folder >}}
{{< /filetree/container >}}
```